### ISSUE TITLE
Build failure with GHC 7.10

### ISSUE THREAD

**Issue by snoyberg** [BioHaskell/hPDB-examples#1]
```
Unpacking to hPDB-examples-1.2.0.1/
Resolving dependencies...
Configuring hPDB-examples-1.2.0.1...
Building hPDB-examples-1.2.0.1...
Preprocessing executable 'PDB2Fasta' for hPDB-examples-1.2.0.1...
[1 of 1] Compiling Main             ( examples/PDB2Fasta.hs, dist/build/PDB2Fasta/PDB2Fasta-tmp/Main.o )
Linking dist/build/PDB2Fasta/PDB2Fasta ...
Preprocessing executable 'ShiftToCenter' for hPDB-examples-1.2.0.1...
[1 of 1] Compiling Main             ( examples/ShiftToCenter.hs, dist/build/ShiftToCenter/ShiftToCenter-tmp/Main.o )

examples/ShiftToCenter.hs:20:1:
    Non type-variable argument in the constraint: Iterable a Atom
    (Use FlexibleContexts to permit this)
    When checking that ‘shift’ has the inferred type
      shift :: forall a. Iterable a Atom => Vector3 -> a -> a

examples/ShiftToCenter.hs:30:11:
    No instance for (PDBWritable a0) arising from a use of ‘write’
    The type variable ‘a0’ is ambiguous
    Note: there are several potential instances:
      instance PDBWritable Atom -- Defined in ‘Bio.PDB.StructurePrinter’
      instance PDBWritable Chain -- Defined in ‘Bio.PDB.StructurePrinter’
      instance PDBWritable Model -- Defined in ‘Bio.PDB.StructurePrinter’
      ...plus two others
    In a stmt of a 'do' block: write s1 outfname
    In the expression:
      do { [inpfname, outfname] <- getArgs;
           Just structure <- parse inpfname;
           let c@(Vector3 x y z) = center structure;
           printf "Center %.2f %.2f %.2f\n" x y z;
           .... }
    In an equation for ‘main’:
        main
          = do { [inpfname, outfname] <- getArgs;
                 Just structure <- parse inpfname;
                 let c@(Vector3 x y z) = ...;
                 .... }
```

**Comment by mgajda** [comment]
Fixed as of 1.2.0.2.
Thanks a lot for bug report!!!

**Comment by mgajda** [comment]
Should be fixed and build correctly by now.
I have uploaded fixesfor both hPDB and hPDB-examples and make sure it
builds withGHC 7.10, and latest packages from Hackage (including zlib

## 0.6.1.0, ifdefs should keep itworking for olderversions.)

  Cheers
     Michal

On Thursday, April 30, 2015, Michael Snoyman notifications@github.com
wrote:

> Unpacking to hPDB-examples-1.2.0.1/
> Resolving dependencies...
> Configuring hPDB-examples-1.2.0.1...
> Building hPDB-examples-1.2.0.1...
> Preprocessing executable 'PDB2Fasta' for hPDB-examples-1.2.0.1...
> [1 of 1] Compiling Main             ( examples/PDB2Fasta.hs, dist/build/PDB2Fasta/PDB2Fasta-tmp/Main.o )
> Linking dist/build/PDB2Fasta/PDB2Fasta ...
> Preprocessing executable 'ShiftToCenter' for hPDB-examples-1.2.0.1...
> [1 of 1] Compiling Main             ( examples/ShiftToCenter.hs, dist/build/ShiftToCenter/ShiftToCenter-tmp/Main.o )
> 
> examples/ShiftToCenter.hs:20:1:
>     Non type-variable argument in the constraint: Iterable a Atom
>     (Use FlexibleContexts to permit this)
>     When checking that ‘shift’ has the inferred type
>       shift :: forall a. Iterable a Atom => Vector3 -> a -> a
> 
> examples/ShiftToCenter.hs:30:11:
>     No instance for (PDBWritable a0) arising from a use of ‘write’
>     The type variable ‘a0’ is ambiguous
>     Note: there are several potential instances:
>       instance PDBWritable Atom -- Defined in ‘Bio.PDB.StructurePrinter’
>       instance PDBWritable Chain -- Defined in ‘Bio.PDB.StructurePrinter’
>       instance PDBWritable Model -- Defined in ‘Bio.PDB.StructurePrinter’
>       ...plus two others
>     In a stmt of a 'do' block: write s1 outfname
>     In the expression:
>       do { [inpfname, outfname] <- getArgs;
>            Just structure <- parse inpfname;
>            let c@(Vector3 x y z) = center structure;
>            printf "Center %.2f %.2f %.2f\n" x y z;
>            .... }
>     In an equation for ‘main’:
>         main
>           = do { [inpfname, outfname] <- getArgs;
>                  Just structure <- parse inpfname;
>                  let c@(Vector3 x y z) = ...;
>                  .... }
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/BioHaskell/hPDB-examples/issues/1.

## 

  Pozdrawiam
    Michał
